pire/extra/capture.h

```diff
@@ -136,9 +137,13 @@ class CapturingScanner: public LoadedScanner {
 
 	CapturingScanner() {}
 	CapturingScanner(const CapturingScanner& s): LoadedScanner(s) {}
-	explicit CapturingScanner(Fsm& fsm)
+	explicit CapturingScanner(Fsm& fsm, size_t distance = 0)
 	{
-		fsm.Canonize();
+		if (distance) {
+			fsm = CreateApproxFsm(fsm, distance);
+		} else {
```

Comment:
IMHO
Presence of `fsm.Canonize()` in the `else` branch misleads a reader to think that the approximate fsm in the `if` branch is not converted from a NFA to a DFA.
Maybe it would be better to remove `approxFsm.Canonize()` statement from `CreateApproxFsm()` and to write here something like
```
if (distance) {
        fsm = CreateApproxFsm(fsm, distance);
}
fsm.Canonize();
```